File: Python/Reverse Linked List II.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param head, a ListNode
    # @param m, an integer
    # @param n, an integer
    # @return a ListNode
    def reverseBetween(self, head, m, n):
        dummy = ListNode(0)
        dummy.next=  head 
        first = dummy
        for i in range(m-1):
            first = first.next 
        second = first.next 
        for j in range(n-m):
            temp = first.next 
            first.next = second.next
            second.next = second.next.next
            first.next.next = temp 
        return dummy.next  
    
            
        
            
        
        
        

File: Java/src/Reverse Linked List II.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode reverseBetween(ListNode head, int m, int n) {
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode first = dummy;
        
        for (int i=0; i < m-1; i++){
            first = first.next;
        }
        
        ListNode second = first.next;
        ListNode temp;
        for (int j=0; j< n-m; j++){
            temp = first.next;
            first.next = second.next;
            second.next = second.next.next;
            first.next.next = temp;
        }
         return dummy.next;
        
    }
}